mixer/pkg/config/crd/store.go

```diff
@@ -148,28 +148,31 @@ loop:
 			time.Sleep(s.retryInterval)
 		}
 		retryCount++
-		resources, err := d.ServerResourcesForGroupVersion(apiGroupVersion)
-		if err != nil {
-			log.Debugf("Failed to obtain resources for CRD: %v", err)
-			continue
-		}
-		s.cacheMutex.Lock()
-		for _, res := range resources.APIResources {
-			if _, ok := s.caches[res.Kind]; ok {
+		kindMap := map[string]string{"legacyConfig": apiGroupVersion}
```

Comment:
Can we plumb the map up-top? (i.e. checkAndCreateCaches() takes a map[string][]string that maps apiGroupVersion to kinds).

Then, from the callsite we can pass a map of:
"config.istio.io": <current-kind-map>

This way, this piece of code becomes fully generic w.r.t. apigroups & kinds.

This might include changing some of the interfaces (i.e. config/store/Backend)

Reply:
Ack